cli/submit.py

```diff
@@ -2,27 +2,42 @@
 import json
 import requests
 
+from cli import LSRI_CLIENT_ID
 from ebi_eva_common_pyutils.logger import logging_config
 from getpass import getpass
+from cli.lsri_auth import LSRIAuth
 
 logger = logging_config.get_logger(__name__)
 
 
 class StudySubmitter:
     def __init__(self, ENA_AUTH_URL="https://www.ebi.ac.uk/ena/submit/webin/auth/token",
-                 SUBMISSION_INITIATE_ENDPOINT ="http://www.ebi.ac.uk/eva/v1/submission/initiate/webin"):
+                 SUBMISSION_INITIATE_WEBIN_URL ="http://www.ebi.ac.uk/eva/v1/submission/initiate/webin",
+                 SUBMISSION_INITIATE_LSRI_URL ="http://www.ebi.ac.uk/eva/v1/submission/initiate/lsri"):
         self.ENA_AUTH_URL = ENA_AUTH_URL
-        self.SUBMISSION_INITIATE_ENDPOINT = SUBMISSION_INITIATE_ENDPOINT
+        self.SUBMISSION_INITIATE_WEBIN_URL = SUBMISSION_INITIATE_WEBIN_URL
+        self.SUBMISSION_INITIATE_LSRI_URL = SUBMISSION_INITIATE_LSRI_URL
 
-    # TODO
     def submit_with_lsri_auth(self):
-        return NotImplementedError
+        logger.info("Proceeding with LSRI authentication...")
+        # For now, it is OK for client ID to be hardcoded because, unlike client secret, it is not sensitive information
+        response = LSRIAuth(client_id=LSRI_CLIENT_ID,
+                            device_authorization_url="https://login.elixir-czech.org/oidc/devicecode",
+                            submission_initiation_url=self.SUBMISSION_INITIATE_LSRI_URL).get_auth_response()
+        if response.status_code == 200:
+            logger.info("LSRI authentication successful!")
+            response_json = json.loads(response.text)
```

Comment:
Eventually we will want to separate the auth from submission. There will be more than one operation that we will want to perform authenticated so we will want to store the response token. 
I also think that some of this like the status code check and token extraction belongs in `LSRIAuth`

Reply:
Created EVA-3337 to address this.